### ISSUE TITLE
More Closely Emulate VB6 IDE Keyboard Code Navigation

### ISSUE THREAD

**Issue by jpbro** [twinbasic/twinbasic#1757]
**Is your feature request related to a problem? Please describe.**
This is not a high priority, but many of us have a lot of muscle memory around keyboard navigation in the IDE and it would be nice to be able to keep it if possible. Right now tB is missing a couple keyboard navigation shortcuts (Ctrl+UpArrow > go to first line after the procedure declaration line of previous procedure, Ctrl+DownArrow > go to first line after the procedure declaration line of next procedure).

It also has slightly different behaviour than VB6 for Shift+F2 - it takes the caret to procedure declaration line in tB, but in VB6 it goes to the first line *after* the procedure declaration line.

**Describe the solution you'd like**
For tB to support Ctrl+UpArrow and Ctrl+DownArrow, and mimic the exact behaviour of the VB6 IDE when pressing Shift+F2 with the caret is in a method call.

**Describe alternatives you've considered**
Taking the time to retrain myself for quick code navigation in tB's IDE.

**Comment by jpbro** [comment]
There's also this undesired behaviour around keyboard navigation:

If you press PgDn/PgUp, the intellisense popup will appear  and capture subsequent UpArrow/DownArrow keypresses which is kind of annoying since I often do big hops around code with PgDn/PgUp and then refine the caret position with Up/Down arrow presses:

![image](https://github.com/twinbasic/twinbasic/assets/5287585/b8db8650-6fa1-451c-9634-0a46f83d5a4e)

**Comment by jpbro** [comment]
Also re: unwanted intellisense visits - pressing Home/End or Ctrl+Home/End will cause the intellisense popup to appear. IMHO Intellisense should not appear for navigation key events, only typing/change events.

**Comment by jpbro** [comment]
> There's also this undesired behaviour around keyboard navigation:
> 
> If you press PgDn/PgUp, the intellisense popup will appear and capture subsequent UpArrow/DownArrow keypresses which is kind of annoying since I often do big hops around code with PgDn/PgUp and then refine the caret position with Up/Down arrow presses:

This issue is mitigated by selecting Legacy Intellisense mode in the _Tools > IDE Options_ menu (thanks for the tip @WaynePhillipsEA!)

**Comment by WaynePhillipsEA** [comment]
Yes, this is a regression.  Was working prior to BETA 424
